src/watertap_contrib/seto/unit_models/surrogate/tests/test_med_tvc_surrogate.py

```diff
@@ -0,0 +1,342 @@
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+    unused_variables_set,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    constraint_scaling_transform,
+    unscaled_variables_generator,
+    unscaled_constraints_generator,
+    badly_scaled_var_generator,
+)
+
+import idaes.logger as idaeslog
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestMEDTVC:
+    @pytest.fixture(scope="class")
+    def MED_TVC_frame(self):
+        # create model, flowsheet
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+        m.fs.water_prop = SeawaterParameterBlock()
+        m.fs.steam_prop = WaterParameterBlock()
+        m.fs.med_tvc = MEDTVCSurrogate(
+            property_package_water=m.fs.water_prop,
+            property_package_steam=m.fs.steam_prop,
+        )
+
+        med_tvc = m.fs.med_tvc
+        feed = med_tvc.feed_props[0]
+        cool = med_tvc.cooling_out_props[0]
+        dist = med_tvc.distillate_props[0]
+        steam = med_tvc.heating_steam_props[0]
+        motive = med_tvc.motive_steam_props[0]
+
+        # System specification
+        # Input variable 1: Feed salinity (30-60 g/L = kg/m3)
+        feed_salinity = 35 * pyunits.kg / pyunits.m**3
+
+        # Input variable 2: Feed temperature (25-35 deg C)
+        feed_temperature = 25
+
+        # Input variable 3: Motive steam pressure (4-45 bar)
+        motive_pressure = 24
+
+        # Input variable 4: System capacity (2,000 - 100,000 m3/day)
+        sys_capacity = 2000 * pyunits.m**3 / pyunits.day
+
+        # Input variable 5: Recovery ratio (30%- 40%)
+        recovery_ratio = 0.3 * pyunits.dimensionless
+
+        feed_flow = pyunits.convert(
+            (sys_capacity / recovery_ratio), to_units=pyunits.m**3 / pyunits.s
+        )  # feed volumetric flow rate [m3/s]
+
+        """
+        Specify feed flow state properties
+        """
+        # Specify feed flow state properties
+        med_tvc.feed_props.calculate_state(
+            var_args={
+                ("flow_vol_phase", "Liq"): feed_flow,
+                ("conc_mass_phase_comp", ("Liq", "TDS")): feed_salinity,
+                ("temperature", None): feed_temperature + 273.15,
+                # feed flow is at atmospheric pressure
+                ("pressure", None): 101325,
+            },
+            hold_state=True,
+        )
+
+        """
+        Specify heating steam state properties
+        """
+        # Flow rate of liquid heating steam is zero
+        steam.flow_mass_phase_comp["Liq", "H2O"].fix(0)
```

Comment:
Hmm.. I wonder if this is something that should be handled within the unit/property models and not when the user goes to build an example.

Reply:
This constraint is moved to the unit model, along with other similar constraints below.